src/NServiceBus.Testing/Invocations.cs

```diff
@@ -16,12 +16,14 @@ abstract class ExpectedInvocation<T> : IExpectedInvocation where T : ActualInvoc
         public void Validate(params ActualInvocation[] invocations)
         {
             var calls = invocations.Where(i => typeof(T) == i.GetType());
-            var success = calls.Any(c =>
-                                         {
-                                             var result = Validate(c as T);
-                                             
-                                             return result;
-                                         });
+
+            var callResults = calls.Select(c =>
+            {
+                var result = Validate(c as T);
+                return result;
+            }).ToArray(); // Force enumeration
```

Comment:
Because we need all of the actions to be invoked. Not forcing enumeration means that when .Any is called, the loop stops after the first true - which is the original bug we're trying to fix.